Assets/HoloToolkit/Utilities/Scripts/SceneLauncherChecker.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    public class SceneLauncherChecker : MonoBehaviour
```

Comment:
What is the purpose of this script? Why use this script instead of including the scene launcher game objects in the SceneLauncher scene directly?

Reply:
Oh, I see. Because SceneLauncher has a SpeechInputSource/Handler its game object becomes DontDestroyOnLoad and if it were in the scene then another instance would be created when you return to the scene. The way I have usually handled that is to make the DontDestroyOnLoad object detect that it isn't the singleton during its Awake function and Destroy itself. Like [this](http://answers.unity3d.com/questions/747603/how-does-dontdestroyonload-work-1.html) or [this](https://honkbarkstudios.com/developer-blog/dontdestroyonload-tutorial/).

Pro: This would keep the SceneLauncher more self-contained in a single script that could be embedded directly into the scene.
Con: A bit weird to make an extra copy only to destroy it.

This is fine, though, if you like the way SceneLauncherChecker currently works.

Reply:
I feel like having it create another instance, only to destroy it is too costly.

Reply:
My guess is that this wouldn't be significant compared with all the other aspects of reloading the scene. You could try measuring it in the profiler to see the impact of creating and destroying the scene launcher game object. (But this is a small point. Although I would probably try the self-destroying approach, the way the SceneLauncherChecker currently works is OK.)